File: go/unique_paths.go
package leetcode

func uniquePaths(m int, n int) int {
	dp := make([]int, n)
	for i := 0; i < len(dp); i++ {
		dp[i] = 1
	}

	for i := 0; i < m-1; i++ {
		for j := 1; j < n; j++ {
			dp[j] = dp[j] + dp[j-1]
		}
	}

	return dp[len(dp)-1]
}

File: python/unique-paths.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        dp = [1]*m
        
        for i in range(n-1):
            for j in range(1, m):
                dp[j] = dp[j]+dp[j-1]
        
        return dp[-1]